packages/components/fields/search-select-field/src/search-select-field.tsx

```diff
@@ -1,320 +1,380 @@
    */
-  loadOptions: PropTypes.func.isRequired,
+  loadOptions: ReactSelectAsyncProps['loadOptions'];
   /**
    * The text shown while the options are being loaded
    */
-  loadingMessage: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+  loadingMessage?: string | (() => string);
   /**
    * If cacheOptions is truthy, then the loaded data will be cached. The cache will remain until cacheOptions changes value.
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  cacheOptions: PropTypes.any,
+  cacheOptions?: ReactSelectAsyncProps['cacheOptions'];
   /**
    * Custom method to filter whether an option should be displayed in the menu
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  filterOption: PropTypes.func,
+  filterOption?: ReactSelectAsyncProps['filterOption'];
   /**
    * The style of the an option in the dropdown menu. It could be single lined option or an option with more and custom info
    */
-  optionType: PropTypes.oneOf([
-    'single-property',
-    'double-property',
-    'multiple-properties',
-  ]),
+  optionType?: 'single-property' | 'double-property' | 'multiple-properties';
   /**
    * A map of errors. Error messages for known errors are rendered automatically.
    * <br />
    * Unknown errors will be forwarded to `renderError`
    */
-  errors: PropTypes.shape({
-    missing: PropTypes.bool,
-  }),
+  errors?: TFieldErrors;
   /**
    * Called with custom errors. This function can return a message which will be wrapped in an ErrorMessage. It can also return null to show no error.
-   * <br />
-   * Signature: `(key, error) => React.node`
    */
-  renderError: PropTypes.func,
+  renderError?: TErrorRenderer;
   /**
    * Indicates if the value is required. Shows an the "required asterisk" if so.
    */
-  isRequired: PropTypes.bool,
+  isRequired?: boolean;
   /**
    * Indicates whether the field was touched. Errors will only be shown when the field was touched.
    */
-  touched: (props, ...rest) =>
-    props.isMulti
-      ? PropTypes.arrayOf(PropTypes.bool, ...rest)(props, ...rest)
-      : PropTypes.bool(props, ...rest),
+  touched?: boolean[] | boolean;
   // LabelField
   /**
    * Title of the label
    */
-  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
+  title: ReactNode;
   /**
    * Hint for the label. Provides a supplementary but important information regarding the behaviour of the input (e.g warn about uniqueness of a field, when it can only be set once), whereas `description` can describe it in more depth. Can also receive a `hintIcon`.
    */
-  hint: requiredIf(
-    PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
-    (props) => props.hintIcon
-  ),
+  hint?: ReactNode;
   /**
    * Provides a description for the title.
    */
-  description: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  description?: ReactNode;
```

Comment:
```suggestion
  description?: string | ReactNode;
```
Any reason why you are defining it as just `ReactNode` and not `string | ReactNode` ?  I see you have same thing on `hint` types too.

Reply:
Remember that [ReactNode](https://github.com/facebook/react/blob/82c8fa90be86fc0afcbff2dc39486579cff1ac9a/packages/shared/ReactTypes.js#L10) type already includes `string` type.

I find [this documentation](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/basic_type_example#useful-react-prop-type-examples) very helpful and come back to it often 😄

Reply:
Cool! thanks Carlos!